### PR TITLE
Port `errors` into `sample-kotlin`

### PR DESCRIPTION
**Pull request by pavlospt**
Part of this commit as well, is that I had to restructure `sample-kotlin` navigation like the one available in `sample`. It will make it easier to port Animations as well.

### CODE REVIEW

File: sample-kotlin/src/main/java/com/fblitho/lithoktsample/NavigatableDemoActivity.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import com.fblitho.lithoktsample.demo.DemoListDataModel
+import java.util.Arrays
+
+@SuppressLint("Registered")
+open class NavigatableDemoActivity : AppCompatActivity() {
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+
+    val indices = intent.getIntArrayExtra(INDICES)
```
Comment (passy) [inline sample-kotlin/src/main/java/com/fblitho/lithoktsample/NavigatableDemoActivity.kt]: What about

```kotlin
    intent.getIntArrayExtra(INDICES)?.let { indices ->
      supportActionBar?.setDisplayHomeAsUpEnabled(true)
      setTitleFromIndices(indices)
    }
```
?

File: sample-kotlin/src/main/java/com/fblitho/lithoktsample/NavigatableDemoActivity.kt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    if (indices != null) {
+      supportActionBar?.setDisplayHomeAsUpEnabled(true)
+      setTitleFromIndices(indices)
+    }
+  }
+
+  override fun onOptionsItemSelected(item: MenuItem): Boolean {
+    when (item.itemId) {
```
Comment (passy) [inline sample-kotlin/src/main/java/com/fblitho/lithoktsample/NavigatableDemoActivity.kt]: That seems pretty verbose for a simple comparison? How about this?

```kotlin
    return if (item.itemId == android.R.id.home) {
      NavUtils.navigateUpFromSameTask(this)
      true
    } else {
      super.onOptionsItemSelected(item)
    }
```

File: sample-kotlin/src/main/java/com/fblitho/lithoktsample/NavigatableDemoActivity.kt [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      parentIntent.putExtra(DemoListActivity.INDICES, Arrays.copyOf(indices, indices.size - 1))
+    }
+
+    return parentIntent
+  }
+
+  private fun setTitleFromIndices(indices: IntArray) {
+    var dataModels: List<DemoListDataModel>? = DataModels.DATA_MODELS
+
+    for (i in 0 until indices.size - 1) {
```
Comment (passy) [inline sample-kotlin/src/main/java/com/fblitho/lithoktsample/NavigatableDemoActivity.kt]: Isn't that off by one? I thought `until` excluded the last element whereas `..` included it?

File: sample-kotlin/src/main/java/com/fblitho/lithoktsample/NavigatableDemoActivity.kt [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      dataModels = dataModels?.get(indices[i])?.datamodels
+    }
+
+    val title = dataModels?.get(indices[indices.size - 1])?.name
+    setTitle(title)
```
Comment (passy) [inline sample-kotlin/src/main/java/com/fblitho/lithoktsample/NavigatableDemoActivity.kt]:
```kotlin
    dataModels?.get(indices[indices.size - 1])?.let { title = it.name }
```

**Comment by pavlospt** [comment]
@passy thanks for reviewing it, will update with feedback comments!

File: sample-kotlin/src/main/java/com/fblitho/lithoktsample/errors/DebugErrorComponentSpec.kt [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+ * Renders a throwable as a text with a title and provides a touch callback that logs the throwable
+ * with WTF level.
+ */
+@LayoutSpec
+object DebugErrorComponentSpec {
+
+  private const val TAG = "DebugErrorComponentSpec"
+
+  @ColorInt
+  private val DARK_RED_FRAME = -0x32b6d8
```
[the remark this answers is not in the document]
Comment (pavlospt) [reply sample-kotlin/src/main/java/com/fblitho/lithoktsample/errors/DebugErrorComponentSpec.kt]: Ohh yes, I was thinking of changing those before committing it but forgot :(. Will update it asap!

[the remark this answers is not in the document]
Comment (pavlospt) [reply sample-kotlin/src/main/java/com/fblitho/lithoktsample/errors/DebugErrorComponentSpec.kt]: Done!

**Pull request closed without merging**